upup/pkg/fi/nodeup/command.go

```diff
@@ -269,6 +270,24 @@ func (c *NodeUpCommand) Run(out io.Writer) error {
 		if err != nil {
 			return err
 		}
+
+		modelContext.MachineType, err = getMachineType()
+		if err != nil {
+			return fmt.Errorf("failed to get machine type: %w", err)
+		}
+
+		awsCloud := cloud.(awsup.AWSCloud)
+		// Get the instance type's detailed information.
+		instanceType, err := awsup.GetMachineTypeInfo(awsCloud, modelContext.MachineType)
+		if err != nil {
+			return err
+		}
+
+		if instanceType.GPU {
+			klog.Info("instance supports GPU acceleration")
+			modelContext.GPUVendor = architectures.GPUVendorNvidia
+		}
```

Comment:
Why does this have to be done for every system? Isn't this only needed if Nvidia is enabled in conf?
Seems like a replay of the previous commit. In short, I don't understand why you need autodetection.